dapp_manager/storage.py

```diff
@@ -138,7 +146,28 @@ def pid_file(self) -> Path:
     def archived_pid_file(self) -> Path:
         return self.file_name("_old_pid")
 
-    def file_name(self, name: Union[RunnerFileType, Literal["pid", "_old_pid"]]) -> Path:
+    @property
+    def api(self) -> Optional[str]:
```

Comment:
The idea of file reading each time on property access is a red flag for me. Nobody expects that under property those things are happening.

Reply:
I disagree with a general statement like this one here - but let's discuss it privately...